### PR TITLE
Migration of test_volume_network_ping_timeout.py

### PR DESCRIPTION
**Pull request by nishith-vihar**
Migration of [test_volume_network_ping_timeout.py](https://github.com/gluster/glusto-tests/blob/master/tests/functional/glusterd/test_volume_network_ping_timeout.py)  from glusto to redant.                            
                                                                                 
Updates: #292                                                                   
Fixes: #539                                                                                                         
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>
Signed-off-by: srijan-sivakumar <ssivakum@redhat.com>

### CODE REVIEW

**Comment by nishith-vihar** [comment]
The test case needs arequal checksum to be installed.

**Comment by nik-redhat** [comment]
@srijan-sivakumar any idea how shall we be handling the installation of the `arequal-checksum` tool? I could only find a repo for it and not an external source to download it :thinking:

**Comment by srijan-sivakumar** [comment]
> @srijan-sivakumar any idea how shall we be handling the installation of the `arequal-checksum` tool? I could only find a repo for it and not an external source to download it thinking

Unfortunately I think that is the only way...I don't remember yum or dnf working on it...still we can check once.

**Comment by srijan-sivakumar** [comment]
```console
[root@dhcp42-44 redant]# python3 core/redant_main.py -c ../config.yml -t tests/functional/glusterd/test_volume_network_ping_timeout.py -ll D -l /tmp/dummyredant/
    ____  __________  ___    _   ________
   / __ \/ ____/ __ \/   |  / | / /_  __/
  / /_/ / __/ / / / / /| | /  |/ / / /   
 / _, _/ /___/ /_/ / ___ |/ /|  / / /    
/_/ |_/_____/_____/_/  |_/_/ |_/ /_/     
                                         

✔ Param Handling Success.
✔ Test List built
✔ Log dir creation successful.
✔ Environment setup successful.
✔ Running test case : vol_create_test-rep
✔ Running test case : vol_create_test-dist
✔ vol_create_test-rep Succeeded
✔ Running test case : test_volume_network_ping_timeout-rep
✔ vol_create_test-dist Succeeded
✔ Running test case : test_volume_network_ping_timeout-dist
✔ test_volume_network_ping_timeout-rep Succeeded
✔ Running test case : vol_destroy_test-rep
✔ test_volume_network_ping_timeout-dist Succeeded
✔ Running test case : vol_destroy_test-dist
✔ vol_destroy_test-rep Succeeded
✔ Running test case : vol_create_test-disp
✔ vol_destroy_test-dist Succeeded
✔ Running test case : vol_create_test-dist-rep
✔ vol_create_test-disp Succeeded
✔ Running test case : test_volume_network_ping_timeout-disp
✔ vol_create_test-dist-rep Succeeded
✔ Running test case : test_volume_network_ping_timeout-dist-rep
✔ test_volume_network_ping_timeout-disp Succeeded
✔ Running test case : vol_destroy_test-disp
✔ test_volume_network_ping_timeout-dist-rep Succeeded
✔ Running test case : vol_destroy_test-dist-rep
✔ vol_destroy_test-disp Succeeded
✔ Running test case : vol_create_test-dist-disp
✔ vol_create_test-dist-disp Succeeded
✔ Running test case : test_volume_network_ping_timeout-dist-disp
✔ vol_destroy_test-dist-rep Succeeded
✔ test_volume_network_ping_timeout-dist-disp Succeeded
✔ Running test case : vol_destroy_test-dist-disp
✔ vol_destroy_test-dist-disp Succeeded
Results for Special
Test nature : nonDisruptive
TestName : vol_create_test
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:00:30       |      NA     |
|     dist    |     PASS    |        00:00:30       |      NA     |
|     disp    |     PASS    |        00:00:29       |      NA     |
|   dist-rep  |     PASS    |        00:00:32       |      NA     |
|  dist-disp  |     PASS    |        00:00:23       |      NA     |
+-------------+-------------+-----------------------+-------------+
TestName : vol_destroy_test
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:00:18       |      NA     |
|     dist    |     PASS    |        00:00:17       |      NA     |
|     disp    |     PASS    |        00:00:28       |      NA     |
|   dist-rep  |     PASS    |        00:00:30       |      NA     |
|  dist-disp  |     PASS    |        00:00:16       |      NA     |
+-------------+-------------+-----------------------+-------------+
Results for glusterd
Test nature : nonDisruptive
TestName : test_volume_network_ping_timeout
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:01:10       |      NA     |
|     dist    |     PASS    |        00:01:10       |      NA     |
|     disp    |     PASS    |        00:01:04       |      NA     |
|   dist-rep  |     PASS    |        00:01:30       |      NA     |
|  dist-disp  |     PASS    |        00:01:08       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------+
|               Category               | Count |
+--------------------------------------+-------+
|           Disruptive Tests           |   0   |
|         Non Disruptive Tests         |   3   |
|       Disruptive Tests Skipped       |   0   |
|     Non Disruptive Tests Skipped     |   0   |
|              Tests Ran               |   3   |
|            Tests Skipped             |   0   |
|             Total Tests              |   3   |
|   Disruptive Tests Pass Percentage   |   0   |
| Non Disruptive Tests Pass Percentage | 100.0 |
|        Total Pass Percentage         | 100.0 |
+--------------------------------------+-------+
Total Time Taken : 00 hours 06 minutes 17 seconds
✔ Tearing down successful.
```

File: common/ops/support_ops/io_ops.py [hunk 1]
```diff
@@ -288,6 +288,36 @@ def create_deep_dirs_with_files(self, path: str, dir_start_no: int,
                f"--num-of-files {no_files} {path}")
         return self.execute_command_async(cmd, node)
 
+    def create_files(self, path: str, no_of_files: str, node: str,
```
Comment (aujjwal-redhat) [inline common/ops/support_ops/io_ops.py:291]: We already have a create_files function in io ops.

**Review by aujjwal-redhat — CHANGES_REQUESTED** [review]

File: common/ops/support_ops/io_ops.py (hunk 1 above)
Comment (srijan-sivakumar) [reply common/ops/support_ops/io_ops.py:291]: Ahh..I guess this was a function written before the create_files which we have now in io_ops..let me remove this and use that.

**Comment by srijan-sivakumar** [comment]
```console
(redant-demo) [root@rhsdev-virt05 redant]# python3 core/redant_main.py -c ../rconfig.yml -t tests/functional/glusterd/test_volume_network_ping_timeout.py -ll D
    ____  __________  ___    _   ________
   / __ \/ ____/ __ \/   |  / | / /_  __/
  / /_/ / __/ / / / / /| | /  |/ / / /   
 / _, _/ /___/ /_/ / ___ |/ /|  / / /    
/_/ |_/_____/_____/_/  |_/_/ |_/ /_/     
                                         

✔ Param Handling Success.
✔ Test List built
✔ Log dir creation successful.
✔ Environment setup successful.
✔ Running test case : vol_create_test-rep
✔ Running test case : vol_create_test-dist
✔ vol_create_test-rep Succeeded
✔ Running test case : test_volume_network_ping_timeout-rep
✔ vol_create_test-dist Succeeded
✔ Running test case : test_volume_network_ping_timeout-dist
✔ test_volume_network_ping_timeout-rep Succeeded
✔ Running test case : vol_destroy_test-rep
✔ test_volume_network_ping_timeout-dist Succeeded
✔ Running test case : vol_destroy_test-dist
✔ vol_destroy_test-dist Succeeded
✔ Running test case : vol_create_test-disp
✔ vol_destroy_test-rep Succeeded
✔ Running test case : vol_create_test-dist-rep
✔ vol_create_test-disp Succeeded
✔ Running test case : test_volume_network_ping_timeout-disp
✔ vol_create_test-dist-rep Succeeded
✔ Running test case : test_volume_network_ping_timeout-dist-rep
✔ test_volume_network_ping_timeout-disp Succeeded
✔ Running test case : vol_destroy_test-disp
✔ test_volume_network_ping_timeout-dist-rep Succeeded
✔ Running test case : vol_destroy_test-dist-rep
✔ vol_destroy_test-disp Succeeded
✔ Running test case : vol_create_test-dist-disp
✔ vol_destroy_test-dist-rep Succeeded
✔ vol_create_test-dist-disp Succeeded
✔ Running test case : test_volume_network_ping_timeout-dist-disp
✔ test_volume_network_ping_timeout-dist-disp Succeeded
✔ Running test case : vol_destroy_test-dist-disp
✔ vol_destroy_test-dist-disp Succeeded
Results for Special
Test nature : nonDisruptive
TestName : vol_create_test
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:00:20       |      NA     |
|     dist    |     PASS    |        00:00:20       |      NA     |
|     disp    |     PASS    |        00:00:32       |      NA     |
|   dist-rep  |     PASS    |        00:00:30       |      NA     |
|  dist-disp  |     PASS    |        00:00:26       |      NA     |
+-------------+-------------+-----------------------+-------------+
TestName : vol_destroy_test
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     dist    |     PASS    |        00:00:13       |      NA     |
|     rep     |     PASS    |        00:00:15       |      NA     |
|     disp    |     PASS    |        00:00:16       |      NA     |
|   dist-rep  |     PASS    |        00:00:24       |      NA     |
|  dist-disp  |     PASS    |        00:00:13       |      NA     |
+-------------+-------------+-----------------------+-------------+
Results for glusterd
Test nature : nonDisruptive
TestName : test_volume_network_ping_timeout
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:01:08       |      NA     |
|     dist    |     PASS    |        00:01:08       |      NA     |
|     disp    |     PASS    |        00:01:08       |      NA     |
|   dist-rep  |     PASS    |        00:01:20       |      NA     |
|  dist-disp  |     PASS    |        00:01:16       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------+
|               Category               | Count |
+--------------------------------------+-------+
|           Disruptive Tests           |   0   |
|         Non Disruptive Tests         |   3   |
|       Disruptive Tests Skipped       |   0   |
|     Non Disruptive Tests Skipped     |   0   |
|              Tests Ran               |   3   |
|            Tests Skipped             |   0   |
|             Total Tests              |   3   |
|   Disruptive Tests Pass Percentage   |   0   |
| Non Disruptive Tests Pass Percentage | 100.0 |
|        Total Pass Percentage         | 100.0 |
+--------------------------------------+-------+
Total Time Taken : 00 hours 05 minutes 46 seconds
✔ Tearing down successful.
```

**Review by aujjwal-redhat — APPROVED** [review]

**Pull request merged**